diff -Nru newlib-1.20.0/libgloss/configure.in newlib-1.20.0-new/libgloss/configure.in
--- newlib-1.20.0/libgloss/configure.in	2011-12-16 06:58:40.000000000 +0800
+++ newlib-1.20.0-new/libgloss/configure.in	2012-12-01 20:39:39.719161029 +0800
@@ -34,6 +34,9 @@
 config_libnosys=true
 
 case "${target}" in
+  i[[3456]]86-myos-elf)
+  	AC_CONFIG_SUBDIRS([myos386])
+	;;
   i[[3456]]86-*-elf* | i[[3456]]86-*-coff*)
 	AC_CONFIG_SUBDIRS([i386])
 	;;
diff -Nru newlib-1.20.0/newlib/configure.host newlib-1.20.0-new/newlib/configure.host
--- newlib-1.20.0/newlib/configure.host	2011-11-29 14:33:48.000000000 +0800
+++ newlib-1.20.0-new/newlib/configure.host	2012-12-01 20:12:05.529542455 +0800
@@ -513,6 +513,11 @@
 # THIS TABLE IS ALPHA SORTED.  KEEP IT THAT WAY.
 
 case "${host}" in
+  i[34567]86-myos-elf)
+       syscall_dir=syscalls
+       posix_dir=posix
+#newlib_cflags="-DHAVE_OPENDIR -D_NO_GLOB -D_NO_EXECVE -D_NO_WORDEXP"
+       ;;
   *-*-cygwin*)
        test -z "$cygwin_srcdir" && cygwin_srcdir=`cd ${srcdir}/../winsup/cygwin; pwd`
        export cygwin_srcdir
diff -Nru newlib-1.20.0/newlib/libc/posix/_isatty.c newlib-1.20.0-new/newlib/libc/posix/_isatty.c
--- newlib-1.20.0/newlib/libc/posix/_isatty.c	2008-11-20 04:55:51.000000000 +0800
+++ newlib-1.20.0-new/newlib/libc/posix/_isatty.c	2012-12-03 19:48:40.261280706 +0800
@@ -1,7 +1,7 @@
 /* isatty.c */
 
 /* Dumb implementation so programs will at least run.  */
-
+/*
 #include <sys/stat.h>
 #include <errno.h>
 
@@ -18,4 +18,4 @@
     return 1;
   errno = ENOTTY;
   return 0;
-}
+}*/
